src/segment-loader.js

```diff
@@ -397,6 +431,11 @@ export default class SegmentLoader extends videojs.EventTarget {
       return null;
     }
 
+    if (mediaIndex === playlist.segments.length - 1 &&
+        this.mediaSource_.readyState === 'ended') {
```

Comment:
I've been trying to keep this free of `this` references so the state it depends on is explicit in the parameters. Is it possible to move this out to the caller or add additional info to the calling arguments?